src/Framework/AspNetCoreAnalyzers/test/Http/HeaderDictionaryAddAnalyzerTests.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Testing;
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Verifiers.CSharpAnalyzerVerifier<
+    Microsoft.AspNetCore.Analyzers.Http.HeaderDictionaryAddAnalyzer>;
+
+namespace Microsoft.AspNetCore.Analyzers.Http;
+
+public class HeaderDictionaryAddAnalyzerTests
+{
+    [Fact]
+    public async Task IHeaderDictionary_WithAdd_ReportsDiagnostics()
+    {
+        // Arrange & Act & Assert
+        await VerifyCS.VerifyAnalyzerAsync(@"
+using Microsoft.AspNetCore.Http;
+namespace HeaderDictionaryAddAnalyzerTests;
+public class Program
+{
+    public static void Main()
+    {
+        var context = new DefaultHttpContext();
+        {|#0:context.Request.Headers.Add(""Accept"", ""text/html"")|};
```

Comment:
What's the behavior if the test input is missing the `;`?

Reply:
Removing the semicolon caused `CS1002` to be added to the list of diagnostics. Since this causes the call to `VerifyCodeFixAsync` to fail due to the extra diagnostic, how would you recommend that I should go about verifying this behavior?

Reply:
@david-acker Just add `{|CS1002:|}` at the end of line, ie,:

```
{|#0:context.Request.Headers.Add(""Accept"", ""text/html"")|}{|CS1002:|}
```

Reply:
When the semicolon is missing in this scenario, the node that's retrieved from the source span is of type `GlobalStatementSyntax` instead of `InvocationExpressionSyntax`. I made some tweaks to the code fix logic to account for this and get invocation from of `GlobalStatementSyntax.Statement.Expression` instead, although this feels a bit hacky to me. Is this the right approach or is there a better way to handle this?

Reply:
> When the semicolon is missing in this scenario, the node that's retrieved from the source span is of type `GlobalStatementSyntax` instead of `InvocationExpressionSyntax`. I made some tweaks to the code fix logic to account for this and get invocation from of `GlobalStatementSyntax.Statement.Expression` instead, although this feels a bit hacky to me. Is this the right approach or is there a better way to handle this?

FindNode has a second optional boolean parameter called getInnermostNodeForTie. Have you tried just passing true to that?

Reply:
Ah that's what I was missing. Thanks! That fixed the issue.

I added a test case with a missing semicolon for both the Append and indexer code fixes.